File: Programmers/[12914] 멀리 뛰기/ONE/12914.java
class Solution {
    public long solution(int n) {
        long[] answer = new long[n + 1];

        answer[0] = 1;
        answer[1] = 1;

        for (int i = 2; i <= n; i++)
            answer[i] = (answer[i - 2] + answer[i - 1]) % 1234567;

        return answer[n] % 1234567;
    }
}

File: Programmers/[12914] 멀리 뛰기/KingDonggyu/12914.js
function solution(n) {
  const dp = [null, 1, 2];

  if (n <= 2) {
    return dp[n];
  }

  for (let x = 3; x <= n; x++) {
    dp[x] = (dp[x - 1] + dp[x - 2]) % 1234567;
  }

  return dp[n];
}